### ISSUE TITLE
The model does not have Zero-shot detection capability

### ISSUE THREAD

**Issue by q3yinii** [AILab-CVC/YOLO-World#640]
Hello, I would like to ask if I trained the YOLO World model with 48 visible classes from the Coco dataset and tested it with 17 invisible classes. The training loss can be reduced, but the test result mAP is very small. Is there a problem with my settings?
【Dataload Settings】
instances_val2017_novel.json：Only retain the relevant annotations for novel classes, categories have not been changed in the instances_val2017 file.
novel_classes=("umbrella","cow","cup","bus","keyboard","skateboard","dog","couch","tie","snowboard","sink","elephant","cake","scissors","airplane","cat","knife")
mscoco_unseen_classes.json=[["umbrella"], ["cow"],["cup"], ["bus"],["keyboard"],["skateboard"], ["dog"], ["couch"],["tie"],["snowboard"],
["sink"],["elephant"],["cake"],["scissors"],["airplane"],["cat"],["knife"]]
coco_train_dataset = dict(_delete_=True,
    type='MultiModalDataset',
    dataset=dict(
        type='YOLOv5CocoDataset',
        metainfo=dict(classes=base_classes),
        data_root='data/coco/',
        ann_file='annotations/instances_train2017_base.json',
        data_prefix=dict(img='train2017/'),
        filter_cfg=dict(filter_empty_gt=False, min_size=32)),
    class_text_path='data/texts/mscoco_seen_classes.json',
    pipeline=train_pipeline)

train_dataloader = dict(
    batch_size=train_batch_size_per_gpu,
    collate_fn=dict(type='yolow_collate'),
    dataset=coco_train_dataset)

test_pipeline = [
    *_base_.test_pipeline[:-1],
    dict(type='LoadText'),
    dict(type='mmdet.PackDetInputs',
         meta_keys=('img_id', 'img_path', 'ori_shape', 'img_shape',
                    'scale_factor', 'pad_param', 'texts'))
]
coco_val_dataset = dict(_delete_=True,
    type='MultiModalDataset',
    dataset=dict(
        type='YOLOv5CocoDataset',
        metainfo=dict(classes=novel_classes),
        data_root='data/coco/',
        ann_file='annotations/instances_val2017_novel.json',
        data_prefix=dict(img='val2017/'),
        filter_cfg=dict(filter_empty_gt=False, min_size=32)),
    class_text_path='data/texts/mscoco_unseen_classes.json',
    pipeline=test_pipeline)
【Model Settings 】
yolo_world_v2_l_clip_large.
text_transform = [
    dict(type='RandomLoadText',
         num_neg_samples=(num_classes, num_classes),
         max_num_samples=num_training_classes,
         padding_to_max=True,
         padding_value=''),
    dict(type='mmdet.PackDetInputs',
         meta_keys=('img_id', 'img_path', 'ori_shape', 'img_shape', 'flip',
                    'flip_direction', 'texts'))
]

mosaic_affine_transform = [
    dict(
        type='MultiModalMosaic',
        img_scale=_base_.img_scale,
        pad_val=114.0,
        pre_transform=_base_.pre_transform),
    dict(
        type='YOLOv5RandomAffine',
        max_rotate_degree=0.0,
        max_shear_degree=0.0,
        max_aspect_ratio=100.,
        scaling_ratio_range=(1 - _base_.affine_scale,
                             1 + _base_.affine_scale),
        # img_scale is (width, height)
        border=(-_base_.img_scale[0] // 2, -_base_.img_scale[1] // 2),
        border_val=(114, 114, 114))
]
train_pipeline = [
    *_base_.pre_transform,
    *mosaic_affine_transform,
    dict(
        type='YOLOv5MultiModalMixUp',
        prob=_base_.mixup_prob,
        pre_transform=[*_base_.pre_transform,
                       *mosaic_affine_transform]),
    *_base_.last_transform[:-1],
    *text_transform
]
train_pipeline_stage2 = [
    *_base_.train_pipeline_stage2[:-1],
    *text_transform
]
【Test results】
<img width="554" height="223" alt="Image" src="https://github.com/user-attachments/assets/031e2137-a688-4f10-abd4-799bf82ec63d" />

**Comment by wangquanquan11** [comment]
> 你好，我想问一下我是否用Coco数据集中的48个可见类训练YOLO World模型，并用17个隐形类测试。训练损失可以减少，但测试结果mAP非常小。我的设置有问题吗？【数据加载设置】 instances_val2017_novel.json：仅保留新颖类的相关注释，instances_val2017文件中的类别未更改。 novel_classes=（“伞”、“牛”、“杯子”、“公交车”、“键盘”、“滑板”、“狗”、“沙发”、“领带”、“单板滑雪”、“水槽”、“大象”、“蛋糕”、“剪刀”、“飞机”、“猫”、“刀”）mscoco_unseen_classes.json=[[“伞”]、[“牛”]、[“杯子”]、[“巴士”]、[“键盘”]、[“滑板”]、[“狗”]、[“沙发”]、[“领带”]、[“雪布”]=[[“伞”]、[“奶牛”]、[“杯子”]、[“巴士”]、[“键盘”]、[“滑板”]、[“狗”]、[“沙发”]、[“领带”]、[“雪雪”]oard“]， [”sink“]，[”elephant“]，[”cake“]，[”scissors“]，[”airplane“]，[”cat“]，[”knife“]] coco_train_dataset = dict（_delete_=True， type='MultiModalDataset'， dataset=dict（ type='YOLOv5CocoDataset'， metainfo=dict（classes=base_classes）， data_root='data/coco/'， ann_file='annotations/instances_train2017_base.json'， data_prefix=dict（img='train2017/'）， filter_cfg=dict（filter_empty_gt=False， min_size=32））， class_text_path='data/texts/mscoco_seen_classes.json'， pipeline=train_pipeline）
> 
> train_dataloader = dict（ batch_size=train_batch_size_per_gpu， collate_fn=dict（类型='yolow_collate'）， dataset=coco_train_dataset）
> 
> test_pipeline = [ _*base_.test_pipeline[：-1]， dict（type='LoadText'）， dict（type='mmdet.PackDetInputs'， meta_keys=（'img_id'， 'img_path'， 'ori_shape'， 'img_shape'， 'scale_factor'， 'pad_param'， 'texts'）） ] coco_val_dataset = dict（_delete_=True， type='MultiModalDataset'， dataset=dict（ type='YOLOv5CocoDataset'， metainfo=dict（classes=novel_classes）， data_root='data/coco/'， ann_file='annotations/instances_val2017_novel.json'， data_prefix=dict（img='val2017/'）， filter_cfg=dict（filter_empty_gt=False， min_size=32））， class_text_path='data/texts/mscoco_unseen_classes.json'， pipeline=test_pipeline） 【Model Settings 】 yolo_world_v2_l_clip_large. text_transform = [ dict（type='RandomLoadText'， num_neg_samples=（num_classes， num_classes）， max_num_samples=num_training_classes， padding_to_max=True， padding_value=''）， dict（type='mmdet.PackDetInputs'， meta_keys=（'img_id'， 'img_path'， 'ori_shape'， 'img_shape'， 'flip'， 'flip_direction'， 'texts'）） ]
> 
> mosaic_affine_transform = [ dict（ type（ type='MultiModalMosaic'， img_scale=_base_.img_scale， pad_val=114.0， pre_transform=_base_.pre_transform）， dict（ type（ type='YOLOv5RandomAffine'， max_rotate_degree=0.0， max_shear_degree=0.0， max_aspect_ratio=100.， scaling_ratio_range=（1 - _base_.affine_scale， 1 + _base_.affine_scale）， # img_scale 是（宽度，高度） 边界=（-_base_.img_scale[0] // 2， -_base_.img_scale[1] // 2）， border_val=（114， 114， 114）） ] train_pipeline = [ *_base_.pre_transform， _mosaic_affine_transform， dict（ type='YOLOv5MultiModalMixUp'， prob=_base_.mixup_prob， pre_transform=[__基数_.pre_transform，*mosaic_affine_transform]）， *_基数_.last_transform[：-1]， *text_transform ] train_pipeline_stage2 = [ *_基数_.train_pipeline_stage2[：-1]， *text_transform ] [测试结果] <img alt="图像" width="554" height="223" src="https://private-user-images.githubusercontent.com/99492201/505696440-031e2137-a688-4f10-abd4-799bf82ec63d.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQ2NjM0MDUsIm5iZiI6MTc2NDY2MzEwNSwicGF0aCI6Ii85OTQ5MjIwMS81MDU2OTY0NDAtMDMxZTIxMzctYTY4OC00ZjEwLWFiZDQtNzk5YmY4MmVjNjNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEyMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMjAyVDA4MTE0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNjNzhiOGRkZjFlNWFiZmFlOTRmOWI0Mjk2ZjQxNTI0NWFkMTExNTE0OGNmYzY3ZGE2YWYwZjVlZjE4NzFhOWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.uZoYwaQGmFTc3GxD838YtSJA-aeoFgYVn96qPskzXDY">

Hello, may I ask if your issue has been resolved? If so, could you share how you solved it?
